packages/mdc-tab-bar/index.ts

```diff
@@ -0,0 +1,175 @@
+
+class MDCTabBar extends MDCComponent<MDCTabBarFoundation> {
+  static attachTo(root: Element): MDCTabBar {
+    return new MDCTabBar(root);
+  }
+
+  private tabList_!: MDCTab[]; // assigned in initialize()
+  private tabScroller_!: MDCTabScroller | null; // assigned in initialize()
+  private handleTabInteraction_!: CustomEventListener<TabInteractionEvent>; // assigned in initialSyncWithDOM()
+  private handleKeyDown_!: SpecificEventListener<'keydown'>; // assigned in initialSyncWithDOM()
+
+  set focusOnActivate(focusOnActivate: boolean) {
+    this.tabList_.forEach((tab) => tab.focusOnActivate = focusOnActivate);
+  }
+
+  set useAutomaticActivation(useAutomaticActivation: boolean) {
+    this.foundation_.setUseAutomaticActivation(useAutomaticActivation);
+  }
+
+  initialize(
+      tabFactory: TabFactory = (el) => new MDCTab(el),
+      tabScrollerFactory: TabScrollerFactory = (el) => new MDCTabScroller(el),
+  ) {
+    this.tabList_ = this.instantiateTabs_(tabFactory);
+    this.tabScroller_ = this.instantiateTabScroller_(tabScrollerFactory);
+  }
+
+  initialSyncWithDOM() {
+    this.handleTabInteraction_ = (evt) => this.foundation_.handleTabInteraction(evt);
+    this.handleKeyDown_ = (evt) => this.foundation_.handleKeyDown(evt);
+
+    this.listen(MDCTabFoundation.strings.INTERACTED_EVENT, this.handleTabInteraction_);
+    this.listen('keydown', this.handleKeyDown_);
+
+    for (let i = 0; i < this.tabList_.length; i++) {
+      if (this.tabList_[i].active) {
+        this.scrollIntoView(i);
+        break;
+      }
+    }
+  }
+
+  destroy() {
+    super.destroy();
+    this.unlisten(MDCTabFoundation.strings.INTERACTED_EVENT, this.handleTabInteraction_);
+    this.unlisten('keydown', this.handleKeyDown_);
+    this.tabList_.forEach((tab) => tab.destroy());
+
+    if (this.tabScroller_) {
+      this.tabScroller_.destroy();
+    }
+  }
+
+  getDefaultFoundation(): MDCTabBarFoundation {
+    // tslint:disable:object-literal-sort-keys
+    return new MDCTabBarFoundation({
+      scrollTo: (scrollX) => this.tabScroller_!.scrollTo(scrollX),
+      incrementScroll: (scrollXIncrement) => this.tabScroller_!.incrementScroll(scrollXIncrement),
+      getScrollPosition: () => this.tabScroller_!.getScrollPosition(),
+      getScrollContentWidth: () => this.tabScroller_!.getScrollContentWidth(),
+      getOffsetWidth: () => (this.root_ as HTMLElement).offsetWidth,
+      isRTL: () => window.getComputedStyle(this.root_).getPropertyValue('direction') === 'rtl',
+      setActiveTab: (index) => this.foundation_.activateTab(index),
+      activateTabAtIndex: (index, clientRect) => this.tabList_[index].activate(clientRect),
+      deactivateTabAtIndex: (index) => this.tabList_[index].deactivate(),
+      focusTabAtIndex: (index) => this.tabList_[index].focus(),
+      getTabIndicatorClientRectAtIndex: (index) => this.tabList_[index].computeIndicatorClientRect(),
+      getTabDimensionsAtIndex: (index) => this.tabList_[index].computeDimensions(),
```

Comment:
is it possible that we have `!tabList_.length`? Should this adapter method also return undefined?

Reply:
Good question. It looks like `getTabDimensionsAtIndex()` only gets called when there's at least one tab, so I don't think this should be a problem.